### PR TITLE
updating to new root version

### PR DESCRIPTION
**Pull request by lhuth**
Fix for issue #649 
to be tested on root 6.22 @herkerta

### CODE REVIEW

**Comment by lhuth** [comment]
Tested with root 6.10 - good to go

File: monitors/onlinemon/src/EUDAQMonitorCollection.cc [hunk 1]
```diff
@@ -32,7 +32,7 @@ void EUDAQMonitorCollection::Write(TFile *file) {
     std::cerr << "EUDAQMonitorCollection::Write File pointer is NULL" << endl;
     exit(-1);
   }
-  if (gDirectory != NULL) // check if this pointer exists
+  if (gDirectory) // check if this pointer exists
```
Comment (simonspa) [inline monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: `!= nullptr`?

File: monitors/onlinemon/src/MonitorPerformanceCollection.cc [hunk 2]
```diff
@@ -25,7 +25,7 @@ void MonitorPerformanceCollection::Write(TFile *file) {
     cout << "MonitorPerformanceCollection::Write File pointer is NULL" << endl;
     exit(-1);
   }
-  if (gDirectory != NULL) // check if this pointer exists
+  if (gDirectory) // check if this pointer exists
```
Comment (simonspa) [inline monitors/onlinemon/src/MonitorPerformanceCollection.cc:28]: same

File: monitors/onlinemon/src/EUDAQMonitorCollection.cc (hunk 1 above)
Comment (lhuth) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: Nope, not possible since root 6.24.

**Review by herkerta — APPROVED** [review]

File: monitors/onlinemon/src/EUDAQMonitorCollection.cc (hunk 1 above)
Comment (simonspa) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: because it's not a pointer?

Comment (simonspa) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: Looking at the redefinition, a

```cpp
gDirectory() != nullptr
```

should work

Comment (lhuth) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: They have wrapped this into some construction to be threadsave, which overrides the operator != twice. 
TDirectory* dir = nullptr
If(gDirectory!=dir)

Would work, but they have a direct check implemented now 🤷

Comment (simonspa) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: Whatever... :)

**Review by simonspa — APPROVED** [review]

File: monitors/onlinemon/src/EUDAQMonitorCollection.cc (hunk 1 above)
Comment (lhuth) [reply monitors/onlinemon/src/EUDAQMonitorCollection.cc:35]: It works....

**Pull request merged**